airflow/cli/commands/dag_command.py

```diff
@@ -497,6 +524,56 @@ def dag_test(args, session=None):
             print(dot_graph.source)
 
 
+def _run_task(ti: TaskInstance, session):
+    """
+    Run a single task instance, and push result to Xcom for downstream tasks. Bypasses a lot of
+    extra steps used in `task.run` to keep our local running as fast as possible
+    Args:
+        ti: TaskInstance to run
+    """
+    log.info("*****************************************************")
+    log.info("Running task %s", ti.task_id)
+    try:
+        ti._run_raw_task(session=session)
```

Comment:
I think is the missing part.
```suggestion
        ti._run_raw_task(session=session)
        session.flush()
```